newm-server/src/main/kotlin/io/newm/server/features/song/repo/SongRepositoryImpl.kt

```diff
@@ -316,6 +352,25 @@ internal class SongRepositoryImpl(
         return transaction.transactionCbor.toByteArray().toHexString()
     }
 
+    override suspend fun updateSongMintingStatus(songId: UUID, mintingStatus: MintingStatus) {
```

Comment:
Much cleaner to do this here than in the QQS Receiver

Reply:
Yes. I also figured there would be some instances where Artist Portal requests would be the thing moving to the next state of the minting state machine so it made sense to move it here.